pulp_ansible/app/serializers.py

```diff
@@ -843,11 +847,18 @@ def create(self, validated_data):
 
                                     if verification_result:
                                         cv_signature = CollectionVersionSigstoreSignature(
-                                            sigstore_bundle=bundle_data,
+                                            sigstore_bundle=sigstore_bundle.to_json(),
                                             signed_collection=cv_instance,
                                             sigstore_signing_service=None,
                                         )
                                         cv_signature.save()
+                                    
+                                    io.seek(0)
+
+                        elif settings.ANSIBLE_SIGNATURE_REQUIRE_VERIFICATION:
+                            raise VerificationFailureException(
+                                "Failed to verify Sigstore signature against all verifying services configured on this repository with ANSIBLE_SIGNATURE_REQUIRE_VERIFICATION enabled."
+                            )
```

Comment:
Not sure this elif statement is on the right indentation. I was expecting it to be a part of the for loop if we don't break out of it  (which I think you need to add since we don't want to save the signature twice if we have a verifier).